src/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -33,7 +33,7 @@ internal static bool IsLock(object obj)
             // CORERT-TODO
             return obj is Lock;
 #else
-            return obj.EETypePtr == typeof(Lock).TypeHandle.ToEETypePtr();
+            return obj.EETypePtr == EETypePtr.EETypePtrOf<Lock>();
```

Comment:
I tried removing the CORERT-TODO here, but the intrinsic is still not low level enough and we end up recursing infinitely from the LdToken helper :/